src/main/webapp/js/finscholar/scholarshippageview/scholarshippageview.js

```diff
@@ -16,50 +16,25 @@
 
 goog.module('finscholar.scholarshippageview');
 
-const GoogDom = goog.require('goog.dom');
-const {scholarshippage} = goog.require('example.templates.scholarshippageviews');
 const {ScholarshipDataHandler} = goog.require('datahandlers.scholarshipdatahandler');
+const {SinglePageView} = goog.require('finscholar.singlepageview');
+const {scholarshippage} = goog.require('example.templates.scholarshippageviews');
 
 /**
  * Class for scholarship page view.
  * @public
  */
-class ScholarshipPageView {
+class ScholarshipPageView extends SinglePageView {
 
   /**
    * @constructor
    */
   constructor() {
-    /** 
-     * @private @const {!ScholarshipDataHandler} dataHandler_ 
-     * The object fetches and formats scholarship data.
-     */
-    this.dataHandler_ = new ScholarshipDataHandler();
-  }
+    super(scholarshippage);
 
-  /**
-   * Render the scholarship page.
-   * @public
-   * @param {!Element} container - The DOM element for single scholarship page.
-   */
-  async renderScholarship(container) {
-    // In the prototype, the id is set to 0 by default. Later we'll pass in id as parameter.
-    const id = 0;
-    let scholarshipData = undefined;
-    try {
-      scholarshipData = await this.dataHandler_.fetchAndFormatSingleScholarshipData(id);
-    } catch (e) {
-      console.log(e);
-      // Throws the error to the caller, and the caller will render an error page instead.
-      throw new Error(`Cannot get data for scholarship ${id}, message: ${e}`);
-    }
-    try {
-      container.innerHTML = scholarshippage({scholarship: scholarshipData});      
-    } catch(e) {
-      console.log(e);
-      // Throws the error to the caller, and the caller will render an error page instead.
-      throw new Error(`Failed to generate html: ${e}`);
-    }
+    this.dataHandler = new ScholarshipDataHandler();
```

Comment:
Don't use a setter. Pass this into the constructor if you want to keep it as a separate class.